### DISCUSSION TITLE
Creazione Proforma per EU

### DISCUSSION THREAD

**Discussion by stefanosilvestrinimia** [fattureincloud/api#337] [Q&A]
ciao,
sto cercando di creare una proforma per il mercato EU. 

Ad esempio:

```
[name] => ACME 111111
[type] => 
[first_name] => Pippo
[last_name] => Pluto
[contact_person] => 
[vat_number] => 43344334334
[tax_code] => 
[address_street] => via roma 10
[address_postal_code] => 99999
[address_city] => Roma
[address_province] => Roma
[address_extra] => 
[country] => Denmark
[country_iso] => dk
[email] => test@test.com
```

Questo non viene validato e ritorna:
```
[country] => Denmark
[country_iso] => dk
```

`    [responseBody:protected] => {"error":{"message":"Invalid request.","validation_result":{"data.entity.country":["The data.entity.country field is not valid."],"data.entity.country_iso":["The data.entity.country iso field is not valid."]}}}
`

Idee? grazie

**Comment by stefanosilvestrinimia** [comment]
any help @emajo ? grazie

**Comment by emajo** [comment]
Ciao @stefanosilvestrinimia, la nazione può essere settata compilando uno tra i campi `country` e `country_iso`, chiamando il metodo [List Detailed Countries](https://developers.fattureincloud.it/api-reference#get-/info/detailed_countries) vengono restituite tutte le nazioni con il loro relativi codici, se decide di utilizzare il campo country dovrà passare il `name` che viene restituito da questa chiamata, se decide di utilizzare country_iso allora dovrà inserire l'`iso` restituito da questa chiamata.

Grazie,
Emanuele.

**Comment by stefanosilvestrinimia** [comment]
grazie tutto chiarissimo
